pkg/identityserver/application_access.go

```diff
@@ -96,15 +93,18 @@ func (is *IdentityServer) createApplicationAPIKey(ctx context.Context, req *ttnp
 	if err != nil {
 		return nil, err
 	}
-	key.Key = token
-	events.Publish(evtCreateApplicationAPIKey.NewWithIdentifiersAndData(ctx, req.GetApplicationIds(), nil))
-	err = is.SendContactsEmail(ctx, req, func(data emails.Data) email.MessageData {
-		data.SetEntity(req)
-		return &emails.APIKeyCreated{Data: data, Key: key, Rights: key.Rights}
+	key.Key = ""
+
+	events.Publish(evtCreateApplicationAPIKey.NewWithIdentifiersAndData(ctx, req.GetApplicationIds(), key))
+	go is.notifyInternal(ctx, &ttnpb.CreateNotificationRequest{
```

Comment:
Should this become a worker pool for monitoring purposes ?

Reply:
Probably, but I didn't want to make the diff bigger than strictly needed. I might do that in a follow-up PR.